### DISCUSSION TITLE
Ability to reuse subpatches across different patches - or create a new user op from a subpatch

### DISCUSSION THREAD

**Discussion by jaalibandar** [cables-gl/cables_docs#129] [Feature Requests] [answered]
Hi, the customOp is great to get started with creating new ops, however, it is still a bit intimidating for people who aren't from a dev background - however everybody can create simple subpatches to automate simple tasks instead of writing code - so was just wondering if it might be possible to reuse subpatches across other patches or treat them in a similar fashion as user ops. 
I am aware that is might not always be a good idea in terms of code optimization but nonetheless - seems like a good to have feature. 

For eg: a subpatch to extract colors by index value from the colorpalette op

**Comment by franckaubry** [comment]
A way to create your own functions (with ops) with be one of my top requests as well!
Actually, the first time I encountered subpatches, I just assumed that they turned into a Cables function that you could reuse as much as needed.
At least make it so that if you copy/duplicate a subpatch, you can go into any of the copies for edit and that would be reflected in all of them.

**Reply by steam0r** [reply, marked as answer]
try this with new subpatchops on cables.gl now!
